### ISSUE TITLE
ascend

### ISSUE THREAD

**Issue by ZLT-a-LL** [ultralytics/ultralytics#23621]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

May I ask when it will support the training, inference, and export of Ascend cards? OM models can perform high-speed inference similar to Tensor models

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ZLT-a-LL, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get started, and an Ultralytics engineer will also assist soon 🙂

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
Ascend (CANN/OM) isn’t currently a supported backend for Ultralytics training/inference or a first-class `export(format=...)` target, and we can’t give an ETA, but for OM inference the typical workaround is to export to ONNX and then convert with Ascend’s toolchain; if you share the exact ATC conversion command/requirements (CANN version, `torch_npu`/PyTorch versions, opset, dynamic/static shapes, and any layout/preprocess constraints), we can better scope what an official integration would need—see the [model deployment options](https://docs.ultralytics.com/guides/model-deployment-options/) and our [contributing guide](https://docs.ultralytics.com/help/contributing/).

**Comment by glenn-jocher** [comment]
Hi @ZLT-a-LL, good news — the Ascend export and inference parts of this request have now shipped in Ultralytics. Models can be exported directly to Huawei `.om` format and then loaded for prediction or validation on Ascend hardware:

```bash
yolo export model=yolo26n.pt format=ascend name=Ascend310B4
yolo predict model=yolo26n_ascend_model source=https://ultralytics.com/images/bus.jpg
```

[Ultralytics Platform](https://platform.ultralytics.com) also now offers managed Huawei exports for Ascend310P1, Ascend310P3, Ascend310B1, and Ascend310B4. See the [Huawei Ascend integration guide](https://docs.ultralytics.com/integrations/ascend/) for the complete local and Platform workflows.

This update covers export, inference, and validation; Ascend training and multi-NPU HCCL support remain separate from this release.

![Ultralytics Platform Huawei Ascend export](https://gist.githubusercontent.com/glenn-jocher/ddb7e1146039d729ddb891c6bc261e6b/raw/0af8f891222ccd17f783f8185b7c36f690e5b87d/ultralytics-platform-huawei-ascend-export.png)
